sematic/db/models/job.py

```diff
@@ -0,0 +1,215 @@
+    source_run_id:
+        The id of the run this job is associated with
+    state_name:
+        The name of the state this job is in
+    status_message:
+        A human-readable status message about the job's most recent state.
+    job_type:
+        The job type (ex: worker/driver) of this job.
+    value_serialization:
+        The Sematic serialization of the full external job. This may be more
+        "volatile" over time relative to overall status information. Thus its
+        contents are not extracted into individual fields.
+    status_history_serializations:
+        The Sematic serialization of the job's JobStatus objects over time. Any
+        time the JobStatus object changes in such a way as to make the instances
+        compare as not equal, a new entry will be added to this list.
+        Element 0 is the most recent, element N the oldest.
+    last_updated_epoch_seconds:
+        The time that the job was last updated against the k8s job
+        it represnets, expressed as epoch seconds. Ex: the last time k8s was
+        queried for whether the job was alive. This differs from updated_at
+        in that it relates to the updates against the external resources, while
+        updated_at relates to updates of the DB record. It is in epoch seconds
+        rather than as a datetime object because it will primarily be used in
+        arithmetic operations with time (ex: comparing which value is more recent,
+        amount of elapsed time since last update) rather than for human-readability
+        of absolute time.
+    created_at:
+        The time this record was created in the DB.
+    updated_at:
+        The time this record was last updated in the DB. See documentation in
+        last_updated_epoch_seconds for how this relates to that field.
+    """
+
+    # Q: Why duplicate data that's already in the json of
+    # status_history_serializations and value_serialization as columns?
+    # A: For two reasons:
+    #    1. It gives us freedom to refactor the dataclass for
+    #    JobStatus/ExternalJob/KubernetesExternalJob later to move fields around,
+    #    while allowing the database columns to stay stable.
+    #    2. It allows for more efficient queries on the explicit columns rather than
+    #    requiring json traversal.
+
+    __tablename__ = "jobs"
+
+    id: str = Column(types.String(), primary_key=True)
+    source_run_id: str = Column(  # type: ignore
+        types.String(),
+        nullable=False,
+    )
+    state_name: str = Column(  # type: ignore
+        types.String(),
+        nullable=False,
+    )
+
+    # Why do we need both state_name and is_active? The former is fairly
+    # free-form and subject to change. The latter should be a stable way
+    # to determine whether the job is in a terminal state or not, and
+    # whether it is expected to change.
+    is_active: int = Column(
+        types.Integer, nullable=False  # sqlite doesn't support bool
```

Comment:
Sometimes SQLAlchemy will take care of the mapping (e.g. `types.JSONB`). Isn't it the case for boolean as well? It would be neater to have a real boolean column in the PG DB.